assets/js/handpicked-products.js

```diff
@@ -0,0 +1,248 @@
+			this.setState( { products: [], loaded: true } );
+			return;
+		}
+		apiFetch( {
+			path: addQueryArgs(
+				'/wc-pb/v3/products',
+				getQuery( this.props.attributes, this.props.name )
+			),
+		} )
+			.then( ( products ) => {
+				this.setState( { products, loaded: true } );
+			} )
+			.catch( () => {
+				this.setState( { products: [], loaded: true } );
+			} );
+	}
+
+	getInspectorControls() {
+		const { attributes, setAttributes } = this.props;
+		const { columns, orderby } = attributes;
+
+		return (
+			<InspectorControls key="inspector">
+				<PanelBody
+					title={ __( 'Layout', 'woo-gutenberg-products-block' ) }
+					initialOpen
+				>
+					<RangeControl
+						label={ __( 'Columns', 'woo-gutenberg-products-block' ) }
+						value={ columns }
+						onChange={ ( value ) => setAttributes( { columns: value } ) }
+						min={ wc_product_block_data.min_columns }
+						max={ wc_product_block_data.max_columns }
+					/>
+				</PanelBody>
+				<PanelBody
+					title={ __( 'Order By', 'woo-gutenberg-products-block' ) }
+					initialOpen={ false }
+				>
+					<ProductOrderbyControl
+						setAttributes={ setAttributes }
+						value={ orderby }
+					/>
+				</PanelBody>
+				<PanelBody
+					title={ __( 'Products', 'woo-gutenberg-products-block' ) }
+					initialOpen={ false }
+				>
+					<ProductsControl
+						selected={ attributes.products }
+						onChange={ ( value = [] ) => {
+							const ids = value.map( ( { id } ) => id );
+							setAttributes( { products: ids } );
+						} }
+					/>
+				</PanelBody>
+			</InspectorControls>
+		);
+	}
+
+	renderEditMode() {
+		const { attributes, debouncedSpeak, setAttributes } = this.props;
+		const onDone = () => {
+			setAttributes( { editMode: false } );
+			debouncedSpeak(
+				__( 'Showing product block preview.', 'woo-gutenberg-products-block' )
+			);
+		};
+
+		return (
+			<Placeholder
+				icon={ <IconWidgets /> }
+				label={ __( 'Hand-picked Products', 'woo-gutenberg-products-block' ) }
+				className="wc-block-products-grid wc-block-handpicked-products"
+			>
+				{ __(
+					'Display a selection of hand-picked products in a grid',
+					'woo-gutenberg-products-block'
+				) }
+				<div className="wc-block-handpicked-products__selection">
+					<ProductsControl
+						selected={ attributes.products }
+						onChange={ ( value = [] ) => {
+							const ids = value.map( ( { id } ) => id );
+							setAttributes( { products: ids } );
+						} }
+					/>
+					<Button isDefault onClick={ onDone }>
+						{ __( 'Done', 'woo-gutenberg-products-block' ) }
+					</Button>
+				</div>
+			</Placeholder>
+		);
+	}
+
+	render() {
+		const { setAttributes } = this.props;
+		const { align, columns, editMode } = this.props.attributes;
+		const { loaded, products } = this.state;
+		const classes = [ 'wc-block-products-grid', 'wc-block-handpicked-products' ];
+		if ( columns ) {
+			classes.push( `cols-${ columns }` );
+		}
+		if ( products && ! products.length ) {
```

Comment:
`products.length` is also checked further down in the render logic - should we create a `hasSelectedProducts` variable to re-use?